### DISCUSSION TITLE
Support `stabilityDays` with `regexManagers`

### DISCUSSION THREAD

**Discussion by epage** [renovatebot/renovate#20691] [Request Help] [answered]
### What would you like Renovate to be able to do?

I was hoping to auto-update the MSRV for my project by using `stabilityDays` but I got an error about the field not being supported
```json
  "regexManagers": [
    {
      "fileMatch": [
        "^rust-toolchain\\.toml$",
        "Cargo.toml$",
        "clippy.toml$",
        "\.clippy.toml$",
        "^\.github/workflows/ci.yml$",
        "^\.github/workflows/rust-next.yml$",
      ],
      "matchStrings": [
        "MSRV.*(?<currentValue>\\d+\\.\\d+(\\.\\d+)?)",
        "(?<currentValue>\\d+\\.\\d+(\\.\\d+)?).*MSRV",
      ],
      "depNameTemplate": "rust",
      "packageNameTemplate": "rust-lang/rust",
      "datasourceTemplate": "github-releases",
      "stabilityDays": 126,  // 3 releases * 6 weeks per release * 7 days per week
    }
  ],
```
```
WARN: Config Warning
{
  "configError": {
    "validationSource": ".github/renovate.json5",
    "validationError": "The renovate configuration file contains some invalid settings",
    "validationMessage": "Regex Manager contains disallowed fields: stabilityDays",
    "message": "config-validation",
    "stack": "Error: config-validation\n    at mergeRenovateConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/merge.js:179:23)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async getRepoConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/config.js:11:14)\n    at async initRepo (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/index.js:34:14)\n    at async Object.renovateRepository (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/index.js:45:18)\n    at async renovateRepository (/home/ubuntu/renovateapp/app/worker/index.js:260:26)\n    at async /home/ubuntu/renovateapp/app/worker/index.js:454:5"
  },
  "res": "created"
}
```

### If you have any ideas on how this should be implemented, please tell us here.

I'm assuming at least some data sources, like github releases, would support reporting the published date, allowing this field to be added.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment, marked as answer]
Converted to a discussion. Please default to opening them and not issues.

`stabilityDays` should be configured within a package rule

**Reply by epage** [reply]
Looks like the part that I was missing was that I needed to make a packageRule that referenced the regex manager.  The "answer" provided here does not mention this but instead makes it sound like I was using the wrong tool.

I would recommend **not** being so quick at turning issues into discussions.  I was asking for a feature.  Yes, the feature exists but this shows a lack in the documentation.  For example the [custom manager documentation](https://docs.renovatebot.com/modules/manager/regex/) makes no reference to package rules and out right says the example regex managers will make changes.  The biggest problem I've had with RenovateBot is the documentation which focuses on the little details but does not do a good job talking about how they compose together.  Please look to user reports like this and consider how they might be a fault in the documentation.

As for preferring discussions,  (the ["Discussion" button](https://github.com/renovatebot/renovate/issues) even says it is for if you are unsure.  I would recommend adjusting "Issues" display if the preference is for discussions.

**Reply by rarkins** [reply]
https://github.com/renovatebot/renovate/blob/main/docs/development/help-us-help-you.md#avoid-complaining-about-docs-after-getting-help
